include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h

```diff
@@ -23,3 +23,11 @@
 // A value of "1" means allocators registered in the env will be used. "0" means the allocators created in the session
 // will be used. Use this to override the usage of env allocators on a per session level.
 #define ORT_SESSION_OPTIONS_CONFIG_USE_ENV_ALLOCATORS "session.use_env_allocators"
+
+// Set to 'ORT' (case sensitive) to load an ORT format model.
+// If unset, model type will default to ONNX unless inferred from filename ('.ort' == ORT format) or bytes to be ORT
+#define ORT_SESSION_OPTIONS_CONFIG_LOAD_MODEL_FORMAT "session.load_model_format"
```

Comment:
Something like 
`constexpr const char* kOrtSessionOptionsConfigLoadModelFormat = "..."`?

Or maybe better
```
namespace onnxruntime {
namespace SessionOptionsConfig {
constexpr const char* kLoadModelFormat = "..."}
}
```


---
In reply to: [481322220](https://github.com/microsoft/onnxruntime/pull/4995#discussion_r481322220) [](ancestors = 481322220)

Reply:
You can't have constexpr or namespaces since this file is included by the C header. Something like const char* kOrtSessionOptionsConfigLoadModelFormat = "...";

Reply:
I tried that approach in my PR, which does not work for c++, 

---
In reply to: [481446081](https://github.com/microsoft/onnxruntime/pull/4995#discussion_r481446081) [](ancestors = 481446081,481322220)

Reply:
Either is fine with me but slight preference for the more strong typed version. Can do in a separate PR as it will touch all the existing usage of these keys.

---
In reply to: [481448530](https://github.com/microsoft/onnxruntime/pull/4995#discussion_r481448530) [](ancestors = 481448530)